### PR TITLE
Improved buildEntityUrl method, so we can ask for an entity specific route

### PR DESCRIPTION
**Pull request by juanjol**

### CODE REVIEW

File: src/Behat/Cores/Drupal8.php [hunk 1]
```diff
@@ -230,9 +230,18 @@ public function getLastEntityId($entity_type, $bundle = NULL) {
...
    */
-  public function entityUri($entity_type, $entity) {
-    $path = $entity->toUrl()->getInternalPath();
-    return $path;
+  public function entityUri($entity_type, $entity, $route = 'canonical') {
+    switch($route) {
+      case 'edit':
+      case 'delete':
+      case 'add':
+        $route = $route . '-form';
```
Comment (mistermoper) [inline src/Behat/Cores/Drupal8.php:238]:
```suggestion
        $rel .= '-form';
```

**Review by mistermoper — CHANGES_REQUESTED** [review]

File: src/Behat/Cores/Drupal8.php [hunk 2]
```diff
@@ -230,9 +230,18 @@ public function getLastEntityId($entity_type, $bundle = NULL) {
...
+        break;
+      case NULL:
+        $route = 'canonical';
+        break;
+    }
```
Comment (mistermoper) [inline src/Behat/Cores/Drupal8.php:243]:
```suggestion
       default:
          return return $entity->toUrl()->getInternalPath() . '/' . $route;
    }
```

File: src/Behat/Cores/Drupal8.php [hunk 3]
```diff
@@ -230,9 +230,16 @@ public function getLastEntityId($entity_type, $bundle = NULL) {
...
+  public function entityUri($entity_type, $entity, $route = 'canonical') {
+    switch($route) {
+      case 'edit':
+      case 'delete':
+      case 'add':
+        $rel .= '-form';
+        break;
+      default:
+        return $entity->toUrl()->getInternalPath() . '/' . $route;
+    }
```
Comment (omarlopesino) [inline src/Behat/Cores/Drupal8.php:242]: In this case, if the method takes the default parameter 'canonical', the URL would be 'entity/[id]/canonical'.

Suggestion:
```suggestion
      default:
        $rel = $route;
    }
    return $entity->linkHasTemplate($rel) ? $entity->toUrl($rel)->getInternalPath() :  $entity->toUrl()->getInternalPath() . '/' . $route;
```

File: src/Behat/Cores/Drupal8.php [hunk 4]
```diff
@@ -230,9 +230,16 @@ public function getLastEntityId($entity_type, $bundle = NULL) {
...
    */
-  public function entityUri($entity_type, $entity) {
-    $path = $entity->toUrl()->getInternalPath();
-    return $path;
+  public function entityUri($entity_type, $entity, $route = 'canonical') {
+    switch($route) {
+      case 'edit':
+      case 'delete':
+      case 'add':
+        $rel .= '-form';
```
Comment (omarlopesino) [inline src/Behat/Cores/Drupal8.php:238]: With the latest changes, the $rel variable is not defined anymore and now it is just a concatenation.

```suggestion
        $rel = sprintf('%s-form', $route) ;
```

**Review by omarlopesino — CHANGES_REQUESTED** [review]

**Comment by omarlopesino** [comment]
PR reviewed again @juanjol please check. I think with the latest changes requested this feature will be finally added.

File: src/Behat/Cores/Drupal8.php (hunk 3 above)
Comment (juanjol) [reply src/Behat/Cores/Drupal8.php:242]: It definitely makes more sense and covers the necessary cases. Thank you!

**Comment by juanjol** [comment]
@omarlopesino All changes requested are applied, I think they are fine. If you see fit, we could merge RM. thanks!!

**Comment by omarlopesino** [comment]
thanks, changes looks good to me, I will finish checking this tomorrow

**Review by omarlopesino — APPROVED** [review]

**Comment by omarlopesino** [comment]
I've checked it in a project and it looks good now. Merging.

**Pull request merged**